engine/runtime/src/main/java/org/enso/compiler/SuggestionsCache.java

```diff
@@ -0,0 +1,151 @@
+package org.enso.compiler;
+
+import buildinfo.Info;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.oracle.truffle.api.TruffleLogger;
+import org.bouncycastle.util.encoders.Hex;
+import org.enso.editions.LibraryName;
+import org.enso.interpreter.runtime.EnsoContext;
+import org.enso.polyglot.Suggestion;
+import scala.jdk.CollectionConverters;
+
+import java.io.Serializable;
+import java.nio.ByteBuffer;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.logging.Level;
+
+public final class SuggestionsCache
+    extends Cache<SuggestionsCache.CachedSuggestions, SuggestionsCache.Metadata> {
+
+  private static final String SUGGESTIONS_CACHE_DATA_EXTENSION = ".suggestions";
+  private static final String SUGGESTIONS_CACHE_METADATA_EXTENSION =".suggestions.meta";
+
+  final LibraryName libraryName;
+
+  public SuggestionsCache(LibraryName libraryName) {
+    this.libraryName = libraryName;
+    this.logLevel = Level.FINEST;
+    this.stringRepr = libraryName.toString();
+    this.entryName = libraryName.name();
+    this.dataSuffix = SUGGESTIONS_CACHE_DATA_EXTENSION;
+    this.metadataSuffix = SUGGESTIONS_CACHE_METADATA_EXTENSION;
+  }
+
+  @Override
+  protected byte[] metadata(String sourceDigest, String blobDigest, CachedSuggestions entry) {
+    var mapper = new ObjectMapper();
```

Comment:
It seems that since version 2, the `ObjectMapper` creation is cheap https://stackoverflow.com/questions/18611565/how-do-i-correctly-reuse-jackson-objectmapper#comment27463038_18615839

Reply:
That's not what I experienced. This causes major slowdowns, which is why I also changed it in https://github.com/enso-org/enso/pull/5791

Reply:
That's not what I experienced. This causes major slowdowns, which is why I also changed it in https://github.com/enso-org/enso/pull/5791